app/services/images/generate_social_image_magickally.rb

```diff
@@ -0,0 +1,196 @@
+module Images
+  MEDIUM_FONT_PATH = Rails.root.join("app/assets/fonts/Roboto-Medium.ttf").freeze
+  BOLD_FONT_PATH = Rails.root.join("app/assets/fonts/Roboto-Bold.ttf").freeze
+  TEMPLATE_PATH = Rails.root.join("app/assets/images/social_template.png").freeze
+  ROUNDED_MASK_PATH = Rails.root.join("app/assets/images/rounded_mask.png").freeze
+
+  class GenerateSocialImageMagickally
+    def self.call(resource = nil)
+      new(resource).call
+    end
+
+    def initialize(resource)
+      @resource = resource
+      @logo_url = Settings::General.logo_png
+    end
+
+    def call
+      if @resource.is_a?(Article)
+        @user = @resource.user
+        read_files
+        url = generate_magickally(title: @resource.title,
+                                  date: @resource.readable_publish_date,
+                                  author_name: @user.name,
+                                  color: @user.setting.brand_color1)
+        @resource.update_column(:social_image, url)
+        ## We only need to bust article. All else can fade naturally
+        EdgeCache::BustArticle.call(@resource)
+      elsif @resource.is_a?(User)
+        @user = @resource
+        read_files
+        @resource.articles.published.where(organization_id: nil, main_image: nil).find_each do |article|
+          url = generate_magickally(title: article.title,
+                                    date: article.readable_publish_date,
+                                    author_name: @user.name,
+                                    color: @user.setting.brand_color1)
+          article.update_column(:social_image, url)
+        end
+      else # Organization
+        @user = @resource
+        read_files
+        @resource.articles.published.where(main_image: nil).find_each do |article|
+          url = generate_magickally(title: article.title,
+                                    date: article.readable_publish_date,
+                                    author_name: @user.name,
+                                    color: @user.bg_color_hex)
+          article.update_column(:social_image, url)
+        end
+      end
+    rescue StandardError => e
+      Rails.logger.error(e)
+      Honeybadger.notify(e)
+    end
+
+    private
+
+    def generate_magickally(title: nil, date: nil, author_name: nil, color: nil)
+      result = draw_stripe(color)
+      result = add_logo(result)
+      result = add_text(result, title, date, author_name)
+      result = add_profile_image(result)
+      upload_result(result)
```

Comment:
It would be nice to have a snapshot test for these; the unit tests are nice, but poke at the service object's internals a bit too much (also they aren't a great substitute for an "eye test"). But that's another thing that's better as a follow-up in my opinion since I'm not really sure about the state of golden file testing with Rails/rspec